app/controllers/resources_controller.rb

```diff
@@ -59,20 +59,48 @@ def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    result = assumed_role.allowed_to?(privilege, resource)
+    begin
+      result = assumed_role.allowed_to?(privilege, resource)
+
+      audit_success(privilege, result)
+    rescue => e
+      audit_failure(e, privilege)
```

Comment:
well, this is not right. you should flip the parameters. i am wondering why this was not caught in a test. can you check this?

Reply:
i see, so we don't have any UTs for this class. care to open an issue for this?